query-node/mappings/src/storage.ts

```diff
@@ -161,6 +165,46 @@ async function updateSingleConnectedAsset<T extends Channel | Video>(db: Databas
   }
 }
 
+// removes connection between dataObject and other entities
+async function disconnectDataObjectRelations(db: DatabaseManager, dataObject: DataObject) {
+  await disconnectSingleDataObjectRelation(db, new Channel(), 'avatarPhoto', dataObject)
+  await disconnectSingleDataObjectRelation(db, new Channel(), 'coverPhoto', dataObject)
+
+  await disconnectSingleDataObjectRelation(db, new Video(), 'thumbnailPhoto', dataObject)
+  await disconnectSingleDataObjectRelation(db, new Video(), 'media', dataObject)
+}
+
+async function disconnectSingleDataObjectRelation<T extends Channel | Video>(db: DatabaseManager, type: T, propertyName: string, dataObject: DataObject) {
+  // prepare lookup condition
+  const condition = {
+    where: {
+      [propertyName + 'DataObject']: dataObject
+    }
+  } // as FindConditions<T>
+
+  // in therory the following condition(s) can be generalized `... db.get(type, ...` but in practice it doesn't work :-\
+  const items = type instanceof Channel
+    ? await db.getMany(Channel, condition)
+    : await db.getMany(Video, condition)
```

Comment:
getMany shouldn't be necessary because a dataobject cannot be re-used for more than one asset, so you could get rid of the iteration in next step and simplify.

Reply:
Unnecessary iteration has been removed, and a note about why it is not needed has been added.